### PR TITLE
HDDS-10057. [Snapshot] 'ozone fs -ls' on '.snapshot' dir of a bucket should list only active snapshots

### PR DESCRIPTION
**Pull request by hemantk-12**
## What changes were proposed in this pull request?
Currently when we do `ozone fs -ls` on `.snapshot` hidden dir of a bucket, it throws an error message if bucket contains deleted snapshot which are not be garbage collected as mentioned in HDDS-10057.
In this change, snapshot active check is added to make sure it lists only active snapshots and their content.
 
## What is the link to the Apache JIRA
HDDS-10057

## How was this patch tested?
Updated the unit test and tested it out on local cluster by increasing the run time of SnapshotDeletingService.

### CODE REVIEW

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java [hunk 1]
```diff
@@ -823,8 +824,10 @@ private List<FileStatusAdapter> listStatusBucketSnapshot(
 
     while (snapshotIter.hasNext()) {
       OzoneSnapshot ozoneSnapshot = snapshotIter.next();
-      res.add(getFileStatusAdapterForBucketSnapshot(
-          ozoneBucket, ozoneSnapshot, uri, owner, group));
+      if (SnapshotInfo.SnapshotStatus.SNAPSHOT_ACTIVE.name().equalsIgnoreCase(ozoneSnapshot.getSnapshotStatus())) {
```
Comment (sadanand48) [inline hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java:827]: do we need ignoreCase here or can we skip it?

Comment (xBis7) [reply hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java:827]: I tried it. It also works without ignoreCase.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java [hunk 2]
```diff
@@ -257,41 +258,70 @@ void testCreateSnapshotFailure(String description,
...
+      int res = ToolRunner.run(shell,
+          new String[]{"-deleteSnapshot", BUCKET_PATH, snapshotName1});
+      // Asserts that delete request succeeded
+      Assertions.assertEquals(0, res);
 
-    // Assert that output contains the snapshot key
-    Assertions.assertTrue(listSnapKeyOut
-        .contains(snapshotKeyPath));
+      // Wait for the snapshot to be marked deleted.
+      SnapshotInfo snapshotInfo = ozoneManager.getMetadataManager()
```
Comment (xBis7) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java:297]: Shouldn't that be inside the `waitFor`?

**Review by xBis7 — COMMENTED** [review]
@hemantk-12 Thanks for the patch. I just have 1 comment.

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java (hunk 1 above)
Comment (hemantk-12) [reply hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java]: I did to be more cautious. I removed it.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java (hunk 2 above)
Comment (hemantk-12) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java]: I copied it from other place. Fixed it.
Thanks for catching it.

**Review by xBis7 — APPROVED** [review]
LGTM!

**Comment by hemantk-12** [comment]
Thanks @xBis7 and @sadanand48 for the review.

**Pull request merged**